sdk/nodejs/provider/experimental/analyzer.ts

```diff
@@ -31,6 +31,8 @@ export type PropertyType = "string" | "integer" | "number" | "boolean" | "array"
 export type PropertyDefinition = ({ type: PropertyType } | { $ref: string }) & {
```

Comment:
This question doesn't belong to this PR but I missed reviewing the previous ones... Do we benefit from having separate types like PropertyDefinition in the analyzer module? In my Java implementation, I found that generating schema types directly made it more straightforward. Python may have been different because of the need to do casing translations and mappings. I'm not sure, just curious.